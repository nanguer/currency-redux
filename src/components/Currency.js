import React from "react";
import { add, addToUser, removeFromUser, remove } from "../actions/actions";
import { connect } from "react-redux";

export const ConnectedCurrency = ({
  currency,
  addToUser,
  removeFromUser,
  add,
  remove
}) => {
  const handleSelect = currency => {
    if (!selected) {
      addToUser(currency);
      remove(currency);
    } else {
      add(currency);
      removeFromUser(currency);
    }
  };

  const { name, code, mid, selected } = currency;
  return (
    <li>
      <div>Currency: {name}</div>
      <div>Code: {code}</div>
      <div>Mid: {mid}</div>
      <div>
        <button onClick={() => handleSelect(currency)}>
          {selected ? "Remove" : "Add"}
        </button>
      </div>
    </li>
  );
};

export default connect(null, { addToUser, removeFromUser, add, remove })(
  ConnectedCurrency
);
