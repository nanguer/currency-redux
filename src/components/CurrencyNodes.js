import React from "react";
import Currency from "../components/Currency";

const CurrencyNodes = ({ currencies }) => (
  <>
    {currencies.length > 0 ? (
      currencies.map((currency, i) => {
        return <Currency key={i} currency={currency} />;
      })
    ) : (
      <p>No currencies...</p>
    )}
  </>
);

export default CurrencyNodes;
