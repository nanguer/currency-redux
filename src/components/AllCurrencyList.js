import React, { useEffect } from "react";
import { connect } from "react-redux";
import CurrencyNodes from "../components/CurrencyNodes";
import { checkIfCurrencies } from "../actions/actions";

export const ConnectedAllCurrencyList = ({ currencies, checkIfCurrencies }) => {
  useEffect(() => {
    checkIfCurrencies();
  }, []);

  useEffect(() => {
    localStorage.setItem("currencyList", JSON.stringify(currencies));
  }, [currencies]);

  return (
    <div>
      <h1>Available Currencies</h1>
      <ul>
        <CurrencyNodes currencies={currencies} />
      </ul>
    </div>
  );
};

const mapStateToProps = state => ({
  currencies: state.currencies
});

export default connect(mapStateToProps, { checkIfCurrencies })(
  ConnectedAllCurrencyList
);
