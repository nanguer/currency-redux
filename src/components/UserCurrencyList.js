import React from "react";
import CurrencyNodes from "./CurrencyNodes";
import { connect } from "react-redux";

const ConnectedUserCurrencyList = ({ userCurrencies }) => {
  const userList = (
    <>
      <h1>Following currencies:</h1>
      <ul>
        <CurrencyNodes currencies={userCurrencies} />
      </ul>
    </>
  );
  return (
    <div style={{ textAlign: "end" }}>
      {userCurrencies.length > 0 && userList}
    </div>
  );
};

const mapStateToProps = state => ({
  userCurrencies: state.userCurrencies
});

export default connect(mapStateToProps)(ConnectedUserCurrencyList);
