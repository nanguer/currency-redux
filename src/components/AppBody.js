import React, { useState } from "react";
import AllCurrencyListLoader from "../components/loaders/AllCurrencyLoader";
import UserCurrencyList from "../components/UserCurrencyList";

const AppBody = () => {
  const [showList, setShowList] = useState(false);

  const handleOnClick = () => {
    setShowList(!showList);
  };

  const textAlign = {
    textAlign: "center"
  };

  return (
    <div className="app-body">
      <div>
        <h1 style={textAlign}>Currency Application</h1>
      </div>
      <div style={textAlign}>
        <h3>Please select which currencies to follow from the list</h3>
      </div>
      <div
        className="currency-columns"
        style={{ display: "flex", flexDirection: "row" }}
      >
        <div className="all-currency">
          <button onClick={handleOnClick} style={{ position: "absolute" }}>
            {showList ? "Hide" : "Show"} currency rates
          </button>
          {showList ? <AllCurrencyListLoader /> : null}
        </div>
        <div>
          <UserCurrencyList />
        </div>
      </div>
    </div>
  );
};

export default AppBody;
