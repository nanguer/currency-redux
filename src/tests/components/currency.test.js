import React from "react";
import { shallow } from "enzyme";
import { ConnectedCurrency } from "../../components/Currency";

const rate = {
  currency: "Argentinian Peso",
  code: "ARS",
  mid: 0.011111,
  selected: false
};

test("Should render Currency correctly", () => {
  const wrapper = shallow(<ConnectedCurrency currency={rate} />);
  expect(wrapper).toMatchSnapshot();
});
