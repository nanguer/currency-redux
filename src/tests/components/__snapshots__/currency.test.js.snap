// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Currency correctly 1`] = `
<li>
  <div>
    Currency: 
  </div>
  <div>
    Code: 
    ARS
  </div>
  <div>
    Mid: 
    0.011111
  </div>
  <div>
    <button
      onClick={[Function]}
    >
      Add
    </button>
  </div>
</li>
`;
